### ISSUE TITLE
TEST: CoreFullClusterRestartIT failing

### ISSUE THREAD

**Issue by ywelsch** [elastic/elasticsearch#36429]
Build failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+bwc-tests/191/consoleText

Relevant log lines:

```
[2018-12-10T11:32:52,051][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [node-0] fatal error in thread [elasticsearch[node-0][clusterApplierService#updateTask][T#1]], exiting
java.lang.AssertionError: existing values: [[299 Elasticsearch-7.0.0-SNAPSHOT-e160346 "[search.remote.foo.seeds] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version." "Mon, 10 Dec 2018 11:32:52 GMT", 299 Elasticsearch-7.0.0-SNAPSHOT-e160346 "[search.remote.foo.skip_unavailable] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version." "Mon, 10 Dec 2018 11:32:52 GMT", 299 Elasticsearch-7.0.0-SNAPSHOT-e160346 "[search.remote.foo.seeds] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version." "Mon, 10 Dec 2018 11:32:51 GMT", 299 Elasticsearch-7.0.0-SNAPSHOT-e160346 "[search.remote.foo.skip_unavailable] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version." "Mon, 10 Dec 2018 11:32:51 GMT"]], existing unique values [[[search.remote.foo.skip_unavailable] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version., [search.remote.foo.seeds] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version.]]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ThreadContextStruct.putResponse(ThreadContext.java:528) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ThreadContextStruct.access$1100(ThreadContext.java:403) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.ThreadContext.addResponseHeader(ThreadContext.java:331) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.logging.DeprecationLogger.deprecated(DeprecationLogger.java:313) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.logging.DeprecationLogger.deprecatedAndMaybeLog(DeprecationLogger.java:148) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting.checkDeprecation(Setting.java:470) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting.getRaw(Setting.java:449) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting.lambda$listSetting$33(Setting.java:1208) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting$ListSetting.lambda$new$0(Setting.java:1282) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting$ListSetting.innerGetRaw(Setting.java:1292) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting.getRaw(Setting.java:450) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting$Updater.hasChanged(Setting.java:951) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting$AffixSetting$1.lambda$getValue$2(Setting.java:683) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:?]
	at java.util.stream.DistinctOps$1$2.accept(DistinctOps.java:175) ~[?:?]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177) ~[?:?]
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1603) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312) ~[?:?]
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:735) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[?:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497) ~[?:?]
	at org.elasticsearch.common.settings.Setting$AffixSetting$1.getValue(Setting.java:677) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.Setting$AffixSetting$1.getValue(Setting.java:666) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.AbstractScopedSettings$SettingUpdater.apply(AbstractScopedSettings.java:547) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.AbstractScopedSettings$1.getValue(AbstractScopedSettings.java:283) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.AbstractScopedSettings$1.getValue(AbstractScopedSettings.java:255) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.AbstractScopedSettings$SettingUpdater.updater(AbstractScopedSettings.java:561) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.settings.AbstractScopedSettings.applySettings(AbstractScopedSettings.java:184) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.cluster.service.ClusterApplierService.applyChanges(ClusterApplierService.java:460) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.cluster.service.ClusterApplierService.runTask(ClusterApplierService.java:418) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.cluster.service.ClusterApplierService$UpdateTask.run(ClusterApplierService.java:162) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:660) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:244) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:207) ~[elasticsearch-7.0.0-SNAPSHOT.jar:7.0.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by martijnvg** [comment]
Another failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+bwc-tests/194/console

**Comment by ywelsch** [comment]
More failures (see https://github.com/elastic/elasticsearch/issues/37189). Can someone have a look here?

**Comment by danielmitterdorfer** [comment]
Analyzing the logs provided in #37189, one thing that stands out is that cluster settings get updated multiple times. In [upgraded-node0.log](https://github.com/elastic/elasticsearch/files/2732975/upgraded-node0.log) we can see this:

```
[2019-01-07T02:31:30,983][INFO ][o.e.c.s.ClusterSettings  ] [node-0] updating [search.remote.foo.seeds] from [[]] to [["localhost:9200"]]
[2019-01-07T02:31:30,986][INFO ][o.e.c.s.ClusterSettings  ] [node-0] updating [search.remote.foo.seeds] from [[]] to [["localhost:9200"]]
[2019-01-07T02:31:30,987][INFO ][o.e.c.s.ClusterSettings  ] [node-0] updating [search.remote.foo.seeds] from [[]] to [["localhost:9200"]]
```

And immediately afterwards, the assertion trips complaining that the deprecation headers for `search.remote.foo.seeds` are contained twice. I assume that we see the headers only twice because assertion trips immediately on duplication. If the assertion were not there, I expect to see the same deprecation header three times.

**Comment by jasontedor** [comment]
Closed by #37622
